src/scss/mixins/_alert.scss

```diff
@@ -0,0 +1,18 @@
+@mixin alert(
+    $color: $primary,
+    $padding: 1.2rem,
+    $margin: 1rem 0,
+    $border-radius: 0.4rem
+) {
+    @include smartColor($color);
+    padding: $padding;
+    margin: $margin;
+    border-radius: $border-radius;
+    font-family: $font-family;
+    font-weight: $font-weight;
+    letter-spacing: $letter-spacing;
+    .alert-link {
+        color: oppositeColor($color);
```

Comment:
a simple function which calculates the color luminance : if it's dark gives dark color or light gives light